src/snowflake/snowpark/dataframe.py

```diff
@@ -1519,11 +1518,27 @@ def _union_by_name_internal(
             rattr for rattr in right_output_attrs if rattr not in right_project_list
         ]
 
-        right_child = self._with_plan(
-            Project(right_project_list + not_found_attrs, other._plan)
-        )
+        from snowflake.snowpark import context
```

Comment:
Should we import the context at the top of this file?

Reply:
That will cause circular import error. I'm thinking of improving this in the next sprint.